### ISSUE TITLE
Netmiko - banner_timeout not exiting once SSH banner is detected

### ISSUE THREAD

**Issue by vv0bbLeS** [ktbyers/netmiko#3528]
Hello, I'm connecting to a Palo Alto firewall, and upon receiving the SSH banner error, I increased the banner_timeout parameter value, which solved the issue. However, I notice in my logs that once the SSH banner is detected, I continue to see `read_channel` messages for the duration of the `banner_timeout` value. Only after the `banner_timeout` waiting time has been exhausted will the code proceed with running commands on the firewall.

What I would expect is, once the SSH banner is detected, `_banner_check` in `transport.py` in Paramiko would exit, as there's no longer a need to wait for the SSH banner. Is there some setting that I'm missing that will help `banner_timeout` not wait the entire time if the SSH banner is found?

### Netmiko version

```
netmiko==4.4.0
paramiko==3.5.0
```

### Netmiko device_type (if relevant to the issue)
(Paste `device_type` between quotes below)

```
paloalto_panos
```

### Error Traceback
(Paste the complete traceback of the exception between quotes below)

```
[netmiko-banner-timeout.txt](https://github.com/user-attachments/files/17671839/netmiko-banner-timeout.txt)
```

### Relevant Python code
(Please try to essentialize your Python code to the minimum code needed to reproduce the issue)
(Paste the code between the quotes below)

```python
from netmiko import ConnectHandler

NETMIKO_READ_TIMEOUT_OVERRIDE = 240

device_fw = {
    'device_type': "paloalto_panos",
    'host': "pa-firewall",
    'username': 'player1',
    'password': 'supersecretpassworddontlookplz',
    'banner_timeout': 39,
    'fast_cli': False,
    'read_timeout_override': NETMIKO_READ_TIMEOUT_OVERRIDE,
    'conn_timeout': 53
}

ssh_session = ssh_session = ConnectHandler(**device_fw)
```

**Comment by ktbyers** [comment]
@vv0bbLeS Can you post what you are seeing here?

```
However, I notice in my logs that once the SSH banner is detected, I continue to see read_channel messages for the duration of the banner_timeout value. Only after the banner_timeout waiting time has been exhausted will the code proceed with running commands on the firewall.
```

Also I know Netmiko, in general, doesn't wait for banner_timeout (as otherwise all SSH connection operations would take over 15s to complete, as that is the default banner timeout)...and Netmiko doesn't wait this time (i.e. most SSH operations complete much faster than 15s).

**Comment by vv0bbLeS** [comment]
@ktbyers I tried to attach my log to my post (this is my first github forum post so hopefully I did it right. It gave me a URL where the file was copied to. I can re-attach if needed!)

**Comment by ktbyers** [comment]
I don't see any issue looking at the log files, I think you are misinterpreting what Netmiko is doing. Basically, Netmiko is waiting for your PA device to respond and it is very slow to do so. 

I don't think this is related to the banner timeout.

**Comment by vv0bbLeS** [comment]
@ktbyers ah ok excellent. I had changed the `banner_timeout` value to a few different values (20 and 39), and noticed that each time the logs showed the full `banner_timeout` waiting period was reached before proceeding further, so I was curious as to why that would be.

However, in further tests that I did today, it looks like it just so happened that in those first few tries, the PA response time just so happened to coincide with the `banner_timeout` value I entered. Upon further testing, the `banner_timeout` value doesn't match the PA response time, so it was an unfortunate coincidence on those first few tests that led me down this rabbit hole! Thanks for looking! We can mark this Issue closed.
